CMakeLists.txt

```diff
@@ -201,49 +172,58 @@ endif()
 
 if(UNIX)
     # On unix-like platforms the library is almost always called libz
-   set_target_properties(zlib zlibstatic PROPERTIES OUTPUT_NAME z)
-   if(NOT APPLE)
-     set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
-   endif()
+    set_target_properties(zlib PROPERTIES OUTPUT_NAME z)
+    if(BUILD_SHARED_LIBS AND NOT APPLE)
+        set_target_properties(zlib PROPERTIES LINK_FLAGS "-Wl,--version-script,\"${CMAKE_CURRENT_SOURCE_DIR}/zlib.map\"")
+    endif()
 elseif(BUILD_SHARED_LIBS AND WIN32)
     # Creates zlib1.dll when building shared library version
     set_target_properties(zlib PROPERTIES SUFFIX "1.dll")
 endif()
 
-if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL )
-    install(TARGETS zlib zlibstatic
+if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
+    install(TARGETS zlib
+        EXPORT ${PROJECT_NAME}Config
         RUNTIME DESTINATION "${INSTALL_BIN_DIR}"
         ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
-        LIBRARY DESTINATION "${INSTALL_LIB_DIR}" )
+        LIBRARY DESTINATION "${INSTALL_LIB_DIR}")
+    install(EXPORT ${PROJECT_NAME}Config
+        NAMESPACE zlib:: DESTINATION "${INSTALL_LIB_DIR}/cmake")
+    target_include_directories(zlib PUBLIC $<INSTALL_INTERFACE:${INSTALL_INC_DIR}>)
 endif()
-if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL )
+if(NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL)
     install(FILES ${ZLIB_PUBLIC_HDRS} DESTINATION "${INSTALL_INC_DIR}")
 endif()
-if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL )
+if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
     install(FILES zlib.3 DESTINATION "${INSTALL_MAN_DIR}/man3")
 endif()
-if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL )
+if(NOT SKIP_INSTALL_FILES AND NOT SKIP_INSTALL_ALL)
     install(FILES ${ZLIB_PC} DESTINATION "${INSTALL_PKGCONFIG_DIR}")
 endif()
 
 #============================================================================
-# Example binaries
+# Example binaries and ctests
 #============================================================================
+enable_testing()
 
 add_executable(example test/example.c)
 target_link_libraries(example zlib)
 add_test(example example)
```

Comment:
These tests would ideally only be added based on some condition, either via an option like is done with `BUILD_EXAMPLES` or by only adding the tests if it is the top level project. i.e.

```cmake
if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
    ... add test(s)
endif()
```

When parent projects pull in zlib via `add_subdirectory()`, such conditions allow those projects to avoid having zlib's own tests forcibly added to the test set.

Reply:
I have previously used the variable `BUILD_TESTING`. Replace `enabled_testing()` with `include(CTest)` and allow useds to set `BUILD_TESTING`.
https://cmake.org/cmake/help/v3.13/command/add_test.html?highlight=build_testing

Reply:
`include(CTest)` does more than add the `BUILD_TESTING` option, it also adds a bunch of targets which may not be desirable.

Reply:
done